src/dbnode/encoding/arrow/tile/series_frame_iterator.go

```diff
@@ -74,6 +74,7 @@ func (b *seriesFrameIter) Reset(
 	b.frameStart = start
 	b.frameStep = frameStep
 	b.curr.reset(start, start+frameStep, id, tags)
+	b.recorder.release()
```

Comment:
@arnikola we've run into the issue where datapoints were leaking from one series to another (in cases where series A has block 1, and series B has blocks 2, 3, ..., B would get data from A block 1). Adding this line was just a poor workaround that we did, could you have a look at this (in the original branch of this piece of code)?
I believe @gediminasgu has improved src/dbnode/integration/large_tiles_test.go to reproduce this issue.